src/store.ts

```diff
@@ -105,13 +110,16 @@ export function useStore(selector, options: SelectorOptions = {}): any {
     const newRevNumber = hashFn(result);
     if ((newRevNumber > 0 && revNumber !== newRevNumber) || !isEqual(oldResult, result)) {
       revNumber = newRevNumber;
-      if (options.onUpdate) {
-        options.onUpdate(result);
-      }
       return true;
     }
     return false;
   }
+  if (options.onUpdate) {
+    store.on('before-update', component,  () => {
+      const newValue = selector(store!.state, component.props!);
+      options.onUpdate(newValue);
+    });
+  }
```

Comment:
For me it's important to call `selector` in this case even if there is no `onUpdate` method.
We use have a custom `isEqual` function that requires that it is called at every render (and especially that is never called if there is no render!) in order to properly save the old value for future compares.

Reply:
Note this is also important for `selectCompareUpdate` from OWL, if you don't call it the latest value of scoped `result` won't be saved and the next compare will be wrong. Maybe this handler should write into `result` as well.

Reply:
But are we guaranteed that a render will happen when inside `__notifyComponents`? Isn't the goal of notify specifically for the components to do their compare? By calling the selector and assuming there will be a render (from parent?) we might miss a case where `selectCompareUpdate` will actually be called just after, and fail due to updated `result` value?

Reply:
As per discussed, the only problem is `onUpdate`, `useStore` will be called before the render so what I said is irrelevant.